rfd/0128-aws-integration-tests.md

```diff
@@ -0,0 +1,257 @@
+    aws:
+     - types: ["eks"]
+       # AWS regions to search for resources from
+       regions: ["us-east-1", "us-west-1"]
+       tags:
+         "env": "ci"
+
+```
+
+#### Kubernetes service config
+
+E2E tests will configure the Kubernetes service to watch `kube_cluster` objects
+with label `env:ci` and serve them. The snippet bellow shows the configuration that
+will be used.
+
+```yaml
+kubernetes_service:
+  enabled: "yes"
+  resources:
+  - tags:
+      "env": "ci"
+```
+
+Once the Teleport components are running, the test expects that the Kubernetes
+service starts heartbeat the clusters discovered by the discovery service.
+`tsh kube ls` will be used to monitor the clusters being served by the Kubernetes
+service. Once the cluster shows up in the list, we consider it ready.
+
+After the initial discovery phase, E2E tests will use `tsh` to simulate client connections
+to the cluster and run a simple command to ensure that the connection works and is forwarded as expected.
+`tsh kube login <kube_cluster>` will be used to generate the kubeconfig, and
+`tsh kubectl get service` will be used to test the connection to the cluster.
+
+#### Requirements
+
+This section describes the requirements for the E2E tests for Kubernetes
+access.
+
+- One or more EKS control plane clusters running on AWS. We don't need that the
+  clusters are running any workloads, we only need that their control plane is
+  running and accessible.
+- Discovery service configured to discover the EKS clusters and with List and Describe
+  permissions to the EKS API. [Permissions](https://goteleport.com/docs/kubernetes-access/discovery/aws/#step-13-set-up-aws-iam-credentials)
+- Kubernetes service IAM role configured with the minimum required permissions
+  to generate the short-lived token and forward the requests to the Kubernetes
+  API. [IAM Mapping](https://goteleport.com/docs/kubernetes-access/discovery/aws/#iam-mapping)
+
+Spin up the EKS clusters takes a long time and it's not feasible to do it for
+each test. To speed up the process, we should use an existing EKS cluster that
+is already running, accessible and configured to allow Teleport Kubernetes service
+to access it. This will allow us to run the integration tests without having to
+wait for the EKS cluster to be created. Since we can interact with the EKS API
+and we do not need to run any workloads on the cluster, the existing EKS cluster
+does not need to have dedicated nodes to run workloads. A single control plane
+deployed on a single availability zone is enough. To ensure that the EKS cluster
+is up to date and we don't have to deal with the cluster upgrades having
+major impacts on the development, we will run the Terraform script to destroy
+and recreate the cluster during the weekend. This ensures that the EKS cluster is
+available for the E2E tests during the week (with minor downtime on weekends)
+and that we don't have to deal with cluster upgrades/security patches.
```

Comment:
@reedloden @wadells @jentfoo We're going to be maintaining persistent set of cloud resources in a CI AWS account for these new end-to-end integration tests. The plan is to be re-provisioning them weekly so we don't need to worry about patching. Do you have any suggestions on whats the best way to automate this process? Is this something that can be handled by Spacelift? Or do we just build a separate GHA workflow that runs on cron schedule?

Reply:
cc @jof and @adaadb6

Reply:
We opted to use the [Spacelift schedule task ](https://docs.spacelift.io/concepts/stack/scheduling) to recreate the stack every Sunday at 12:00AM PST time